pyOCD/target/target_kinetis.py

```diff
@@ -141,3 +164,20 @@ def massErase(self):
         else:
             logging.error("Failed to unlock. MDM status: 0x%x", val)
             return False
+
+    ## @brief Write the unsecure FSEC value to flash.
+    #
+    # This will prevent the device from being secured after the next reset.
+    def unlockFlash(self):
+        logging.info("Unlocking chip...")
+
+        # Init flash algorithm.
+        self.flash.init()
+
+        logging.info("Performing mass erase")
+        self.massErase()
```

Comment:
Mass erase actually leaves the FCF blank. Internally, mass erase (however triggered) uses the Erase All Blocks flash command.

From the FTFA chapter's description of the Erase All Blocks command:
If the flash memory module verifies that all flash memories were properly erased, access
control is disabled and security is released by setting the FSEC[SEC] field _[this means the FSEC register in FTFA]_ to the unsecure state. … The security byte and all other contents of the flash configuration field (see Flash Configuration Field Description) are erased by the Erase All Blocks command.

So security is only temporarily disabled until the next reset.

On the KL43 and follow-on L-series devices we added a new Erase All Unsecure flash command that _does_ program the FSEC field of FCF after erasing everything. But this command is not used by mass erase triggered from the MDM-AP.